Add option to Change the created_via field.

scottyzen opened the pull request:
Can be useful when WooCommerce is being used from multiply sources. For example, with plugins like Point of Sale for WooCommerce.

### Your checklist for this pull request
Thanks for sending a pull request! Please make sure you click the link above to view the contribution guidelines, then fill out the blanks below.

🚨Please review the [guidelines for contributing](./CONTRIBUTING.md) to this repository.

- [ ] Make sure you are making a pull request against the **develop branch** (left side). Also you should start *your branch* off *our develop*.
- [ ] Make sure you are requesting to pull request from a **topic/feature/bugfix/devops branch** (right side). Don't pull request from your master!
- [ ] Have you ensured/updated that CLI tests to extend coverage to any new logic. Learn how to modify the tests [here](https://woographql.com/contributing/2-local-testing).

What does this implement/fix? Explain your changes.
---------------------------------------------------
…


Does this close any currently open issues?
------------------------------------------
…


Any relevant logs, error output, GraphiQL screenshots, etc?
-------------------------------------
(If it’s long, please paste to https://ghostbin.com/ and insert the link here.)


Any other comments?
-------------------
…


Where has this been tested?
---------------------------

- **WooGraphQL Version:** 0.10.7
- **WPGraphQL Version:** 1.6.12
- **WordPress Version:** 5.9
- **WooCommerce Version:** 6.2.0

kidunot89 reviewed (changes requested):
I'm fine with this change, but add it to the unit tests [here](https://github.com/wp-graphql/wp-graphql-woocommerce/blob/develop/tests/wpunit/CheckoutMutationTest.php) and [here](https://github.com/wp-graphql/wp-graphql-woocommerce/blob/develop/tests/wpunit/OrderMutationsTest.php)

scottyzen committed:
Add createdVia to unit tests

scottyzen committed:
Add createdVia to unit tests

sbolinger-godaddy commented:
Can this also be added to the checkout mutation? Since the checkout creates an order, the checkout input should accept `createdVia`.

kidunot89 commented:
@scottyzen Get this rebased and we can merge it if it passes CI.

scottyzen committed:
Add option to Change the created_via field.

Can be useful when WooCommerce is being used from multiply sources. For example, with plugins like Point of Sale for WooCommerce.

scottyzen committed:
Rebase

scottyzen committed:
Rebase

kidunot89 commented on tests/wpunit/CheckoutMutationTest.php:
```diff
@@ -85,181 +85,178 @@ function() {
+                                    isCompound
+                                    taxRate {
+                                        databaseId
+                                    }
+                                }
+                            }
+                            lineItems {
+                                nodes {
+                                    productId
+                                    variationId
+                                    quantity
+                                    taxClass
+                                    subtotal
+                                    subtotalTax
+                                    total
+                                    totalTax
+                                    taxStatus
+                                    product {
+                                        ... on SimpleProduct {
+                                            id
+                                        }
+                                        ... on VariableProduct {
+                                            id
+                                        }
+                                    }
+                                    variation {
+                                        id
+                                    }
+                                }
+                            }
+                        }
+                        customer {
+                            id
+                        }
+                        result
+                        redirect
+                    }
+                }
+            ';
+		}
```
@scottyzen What's the purpose of this change? Also you didn't change the `checkout` mutation but the `createOrder` mutation. You should update the `OrderMutationsTest` instead.

kidunot89 commented on tests/wpunit/OrderMutationsTest.php:
```diff
@@ -483,7 +484,7 @@ public function testUpdateOrderMutation() {
 			'paymentMethod'      => 'bacs',
 			'paymentMethodTitle' => 'Direct Bank Transfer',
-			'billing'            => [
+			'billing'            => array(
```
Please revert WooGraphQL uses short array syntax `[ ]`

kidunot89 commented on tests/wpunit/OrderMutationsTest.php:
```diff
@@ -808,7 +809,7 @@ public function testDeleteOrderMutation() {
 			'paymentMethod'      => 'bacs',
 			'paymentMethodTitle' => 'Direct Bank Transfer',
-			'billing'            => [
+			'billing'            => array(
```
Please revert WooGraphQL uses short array syntax `[ ]`

kidunot89 commented on tests/wpunit/OrderMutationsTest.php:
```diff
@@ -959,7 +960,7 @@ public function testDeleteOrderItemsMutation() {
 			'paymentMethod'      => 'bacs',
 			'paymentMethodTitle' => 'Direct Bank Transfer',
-			'billing'            => [
+			'billing'            => array(
```
Please revert WooGraphQL uses short array syntax `[ ]`

kidunot89 reviewed (changes requested):
@scottyzen Your current changes only effect the `createOrder` mutation, you might want to use this functionality in the `checkout` mutation as well. Also, these changes need to be tested in the `CheckoutMutationTest` and `OrderMutationsTest`

scottyzen committed:
Merge remote-tracking branch 'upstream/master' into develop

scottyzen committed:
refactor: Replace array() with short array syntax in OrderMutationsTest

scottyzen committed:
fix: Remove unused 'createdVia' field from CheckoutMutationTest
